packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+
+import { InputIcon } from '../InputIcon';
+import { StateNotifications } from '../StateNotifications';
+import { TogglePassword } from '../TogglePassword';
+
+import {
+  inputBaseStyles,
+  inputBaseThemeStyles,
+  inputDisabledBaseStyles,
+  inputDisabledThemeStyles,
+  inputIconSizeStyles,
+  inputSizeStyles,
+  inputThemeStyles,
+  inputWrapperStyles,
+  labelBaseStyles,
+  labelLargeOverrideStyles,
+} from './PasswordInput.styles';
+import {
+  type NotificationProps,
+  PasswordInputProps,
+  Size,
+  States,
+} from './PasswordInput.types';
+import { getStateFromArray } from './utils';
+
+export const PasswordInput = React.forwardRef<
+  HTMLInputElement,
+  PasswordInputProps
+>(
+  (
+    {
+      value: valueProp,
+      onChange: onChangeProp,
+      darkMode: darkModeProp,
+      id: idProp,
+      'aria-describedby': ariaDescribedbyProp,
+      'aria-labelledby': ariaLabelledbyProp,
+      size = Size.Default,
+      stateNotifications = [],
+      disabled = false,
+      autoComplete = 'new-password',
+      className,
+      label,
+      ...rest
+    }: PasswordInputProps,
+    forwardedRef,
+  ) => {
+    const [showPassword, setShowPassword] = useState<boolean>(false);
+    const prefix = 'lg-passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const descriptionId = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const labelId = useIdAllocator({
+      prefix,
+      id: ariaLabelledbyProp,
+    });
+    const { theme, darkMode } = useDarkMode(darkModeProp);
+    const { value, handleChange } = useControlledValue(valueProp, onChangeProp);
+
+    if (!label && !ariaLabelledbyProp && !rest['aria-label']) {
+      console.warn(
+        'For screen-reader accessibility, label, aria-labelledby, or aria-label must be provided to PasswordInput component',
+      );
+    }
+
+    // If disabled then hide password
+    useEffect(() => {
+      if (disabled) setShowPassword(false);
+    }, [disabled]);
+
+    const handleTogglePasswordClick = () => setShowPassword(s => !s);
+
+    /**
+     * The overall state of the component
+     */
+    const state: States = Array.isArray(stateNotifications)
+      ? getStateFromArray(stateNotifications)
+      : stateNotifications;
+
+    /**
+     * Checks if there are any notifications
+     */
+    const hasNotifications: boolean =
+      !ariaDescribedbyProp && Array.isArray(stateNotifications);
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <div className={className} ref={forwardedRef}>
+          {label && (
+            <Label
+              id={labelId}
+              className={cx(labelBaseStyles, {
+                [labelLargeOverrideStyles]: size === Size.Large,
+              })}
+              htmlFor={inputId}
+              disabled={disabled}
+            >
+              {label}
+            </Label>
+          )}
+          <div className={inputWrapperStyles}>
```

Comment:
Does the input never have a `<Description/>`?

Reply:
no, i confirmed with sooa